localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -169,7 +169,9 @@ def resolve_ref(
     # resource
     resource = resources.get(ref)
     if not resource:
-        raise Exception("Should be detected earlier.")
+        raise Exception(
+            f"Resource target for `!Ref {ref}` could not be found. Is there a resource with name {ref} in your stack?"
```

Comment:
*with the logical resource id

Reply:
I was trying to not be too technical in this message. I think referring to "name" is more actionable, rather than "logical resource id" which is technically correct.